lib/auth.js

```diff
@@ -26,8 +26,10 @@ class Auth {
 
   [kGetAuthDirectiveAST] (astNode) {
     if (Array.isArray(astNode.directives) && astNode.directives.length > 0) {
-      const authDirective = astNode.directives.find(directive => directive.name.value === this[kAuthDirective])
-      if (typeof authDirective !== 'undefined') {
+      const authDirective = astNode.directives.filter(directive => directive.name.value === this[kAuthDirective])
+      if (authDirective.length === 1) {
+        return authDirective[0]
+      } else if (authDirective.length > 1) {
```

Comment:
In order to avoid to introduce breaking changes I've returned an array only when  the directive ie repeted.

I could make this check more smart and return an Array when a directory is repeatable.
This way the getPolicy result would be become more predictable I think.

WDYT?

Reply:
I looked a bit and the `getPolicy` function is used internally, so there should not be any API breaking

https://github.com/mercurius-js/auth/blob/4fc2b1bb276d5faa94faa6e16afd40f03a9ad253/index.js#L16

I think we should test this case: `the user adds the directive multiple times but it is not a repeatable one`. I would expect this case will be already manage by mercurius-core and this module will not have to deal with this case, but it worth testing it

Reply:
Since there are no breaking changes i will also make this function return `Array | null` so everythink would become simpler.

When a non repeatable directive is added multiple times the `graphql/validation` library throws `The directive "@counter" can only be used once at this location.`

Do you think i should add a unit test for this?

Reply:
I think it is worth it.

Note that the CI is failing due to the `.?` syntax

Reply:
Added the new test case and fixed the test for Node 12